### PR TITLE
Fix/erc7739 app domain separator

### PR DESCRIPTION
**Pull request by zeroknots**
added ERC7739 fix + APP_DOMAIN_SEPARATOR logic

### CODE REVIEW

**Comment by filmakarov** [comment]
What do you think of the [new ERC-7739 detection interface](https://github.com/ethereum/ERCs/blob/master/ERCS/erc-7739.md#support-detection)?
Should we remove `supportsNestedTypedDataSign()` since it is [considered deprecated](https://github.com/ethereum/ERCs/blob/master/ERCS/erc-7739.md#detection-of-previous-draft)?

That's how I see it now:
- Account checks if this is the detection request with all the IValidators: [link](https://github.com/bcnmy/nexus/blob/69f437bb5db46e60c62bb780fcebc4120cab130a/contracts/Nexus.sol#L229) [link](https://github.com/bcnmy/nexus/blob/69f437bb5db46e60c62bb780fcebc4120cab130a/contracts/Nexus.sol#L336-L350)
- IValidator should properly reply to detection request : [link](https://github.com/erc7579/erc7739Validator/blob/dfc03c320672c1851890260cc996f4b8e555fa60/src/ERC7739Validator.sol#L35-L48)

I was thinking of using  `IValidator.supportsNestedTypedDataSign` for it, however future IValidators may decide not to implement it. While isValidSignatureWithSender is required by 7579

File: contracts/lib/HashLib.sol [hunk 1]
```diff
@@ -216,10 +178,54 @@ library HashLib {
...
+        return keccak256(
+            abi.encode(
+                EIP712_DOMAIN_TYPEHASH,
+                keccak256(bytes(erc7739Data.name)),
+                keccak256(bytes(erc7739Data.version)),
+                erc7739Data.chainId,
+                erc7739Data.verifyingContract
+            )
+        );
+    }
```
Comment (filmakarov) [inline contracts/lib/HashLib.sol:203]: Do we need both of them?

Comment (zeroknots) [reply contracts/lib/HashLib.sol]: @filmakarov addressed both points in the commits below

File: contracts/SmartSession.sol [hunk 2]
```diff
@@ -381,6 +382,7 @@ contract SmartSession is ISmartSession, SmartSessionBase, SmartSessionERC7739 {
         override
         returns (bytes4 result)
     {
+        if (hash == 0x7739773977397739773977397739773977397739773977397739773977397739) return bytes4(0x77390001);
```
Comment (filmakarov) [inline contracts/SmartSession.sol:385]:
```suggestion
        // ERC-7739 support detection
        if (hash == 0x7739773977397739773977397739773977397739773977397739773977397739) return bytes4(0x77390001);
```

**Review by filmakarov — APPROVED** [review]

**Pull request merged**